src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -155,6 +155,40 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host - host status: {scriptHostManager?.State.ToString()}");
+
+            if (scriptHostManager?.State != ScriptHostState.Running)
```

Comment:
All of these state checks make me nervous about race conditions. For example, drain and resume called really close together so the states haven't been fully updated yet.

To help debug this, it may be a good idea to log when the State changes in the `WebJobsScriptHostService` -- look at what we do for `ActiveHost` today for an example.

Reply:
I added trace logs for the state change - thoughts? https://github.com/Azure/azure-functions-host/pull/7801/commits/54eb08e55313fe18fcb5f59e5b83414c1898c764

RE: synchronizing access - I'd like to bring this topic to our design meeting Thursday, would love to brainstorm with the team on next steps

Reply:
@fabiocav Added semaphore to address this, thoughts?